ggml-backend.c

```diff
@@ -213,6 +213,10 @@ void ggml_backend_tensor_get_async(ggml_backend_t backend, const struct ggml_ten
 }
 
 GGML_CALL void ggml_backend_tensor_set(struct ggml_tensor * tensor, const void * data, size_t offset, size_t size) {
+    if (!size) {
+        return;
+    }
```

Comment:
I think we should do this after the assertions - even in the no-op case we want to be strict about expectations.